packages/react/src/functions/safe-dynamic-require.ts

```diff
@@ -6,5 +6,5 @@ const noop = () => null;
 // Allow us to require things dynamically safe from webpack bundling
 
 export let safeDynamicRequire: typeof require;
-if (typeof globalThis.require === 'function') safeDynamicRequire = eval('require');
+if (Builder.isServer) safeDynamicRequire = eval('require');
```

Comment:
Can we: 
- Understand why the change was made in the first place? Was it when @Dkendal was trying to get the React SDK working in the Hydrogen environment?
- Document this reversal and the difference between both attempts so that we have more information in the future about t these 2 choices?
- 
```suggestion
/* 
 * We initially tried to check the existence of `typeof globalThis.require === 'function'` but that didn't work because of X.
 * This is why we make sure to use `Builder.isServer` to determine whether we can use `require` or not
 */
if (Builder.isServer) safeDynamicRequire = eval('require');
```

Reply:
`Builder.isServer` is not reliable enough to decide to use `eval(...)` - eval is not available in cloudfront workers.

I would also like to know why the typecheck on `globalThis.require` doesn't work as intented, my understanding was that it should function in all environments.

The only way that you can programatically detect that the server code is running in cloudflare workers is if `caches.default` is defined, which is ... less that ideal.

Reply:
Which is super weird, because if I run `globalThis.require` in a Node.JS command prompt, I do get it...

Reply:
where'd the report come from?

Reply:
![image](https://user-images.githubusercontent.com/3162299/192556478-6e048bc8-1f66-4f9a-96ed-14d2b93b4e18.png)
should work assume they are nodejs v12 >

Reply:
Updated as per our discussion in the thread. Added comments for history